src/main/java/com/google/cloud/tools/action/app/DeployAction.java

```diff
@@ -0,0 +1,95 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.action.app;
+
+import com.google.appengine.repackaged.com.google.api.client.util.Strings;
+import com.google.cloud.tools.action.app.config.DeployConfiguration;
+import com.google.cloud.tools.sdk.CloudSdkExecutor;
+import com.google.cloud.tools.sdk.CloudSdkExecutorException;
+import com.google.common.base.Preconditions;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Deploys an app to the server.
+ */
+public class DeployAction {
+
+  private DeployConfiguration configuration;
+  private CloudSdkExecutor sdkExec;
+
+  public DeployAction(DeployConfiguration configuration, CloudSdkExecutor sdkExec) {
+    Preconditions.checkNotNull(configuration);
+    Preconditions.checkNotNull(configuration.getDeployables());
+    Preconditions.checkArgument(configuration.getDeployables().size() > 0);
+    Preconditions.checkNotNull(sdkExec);
+
+    this.configuration = configuration;
+    this.sdkExec = sdkExec;
+  }
+
+  public int execute() throws CloudSdkExecutorException {
+    List<String> arguments = new ArrayList<>();
+    arguments.add("deploy");
+    for (Path deployable : configuration.getDeployables()) {
+      if (!Files.exists(deployable)) {
+        throw new IllegalArgumentException(
+            "Deployable " + deployable.toString() + " does not exist.");
+      }
+      arguments.add(deployable.toString());
+    }
+
+    if (!Strings.isNullOrEmpty(configuration.getBucket())) {
+      arguments.add("--bucket");
+      arguments.add(configuration.getBucket());
+    }
+
+    if (!Strings.isNullOrEmpty(configuration.getDockerBuild())) {
+      arguments.add("--docker-build");
+      arguments.add(configuration.getDockerBuild());
+    }
+
+    if (configuration.isForce()) {
+      arguments.add("--force");
+    }
+
+    if (!Strings.isNullOrEmpty(configuration.getImageUrl())) {
+      arguments.add("--image-url");
+      arguments.add(configuration.getImageUrl());
+    }
+
+    if (!Strings.isNullOrEmpty(configuration.getServer())) {
+      arguments.add("--server");
+      arguments.add(configuration.getServer());
+    }
+
+    if (configuration.isStopPreviousVersion()) {
+      arguments.add("--stop-previous-version");
+    }
+
+    if (!Strings.isNullOrEmpty(configuration.getVersion())) {
+      arguments.add("--version");
+      arguments.add(configuration.getVersion());
+    }
+
+    arguments.add("--quiet");
```

Comment:
I thought of that, but wouldn't it make the code less readable? Now that this code is already written, rewriting it to use reflection doesn't seem like a good use of time.